### DISCUSSION TITLE
[QUESTION] What does the kafka protocol version mean?

### DISCUSSION THREAD

**Discussion by BewareMyPower** [open-telemetry/opentelemetry-collector#3558] [Q&A]
From the https://github.com/open-telemetry/opentelemetry-collector/tree/main/exporter/kafkaexporter#kafka-exporter I see 

> protocol_version (no default): Kafka protocol version e.g. 2.0.0

However, the protocol version is confused. With the 2.0.0 protocol version, the server side received a SaslHandshake request whose version is v0. But when I use a Kafka Java client 2.0.0 to configure SASL_PLAIN authentication, the API version of SaslHandshake request is v1 not v0.

It looks like the protocol version is not what it is for Java client.

What I want to know is:
1. How many protocol versions can I configure?
2. More specifically, how can I configure to send a SaslHandshake request v1?

**Comment by tech4gk** [comment]
@BewareMyPower  It looks like, the Sarama does support header version as 1..
https://github.com/Shopify/sarama/blob/master/sasl_authenticate_request.go#L27
Not sure, why OTEL Kafka Exporter is sending version 0 for header.
